GVFS/GVFS.Platform.Windows/WindowsFileSystemVirtualizer.cs

```diff
@@ -196,6 +196,12 @@ public override FileSystemResult WritePlaceholderDirectory(string relativePath)
             return new FileSystemResult(HResultToFSResult(result), unchecked((int)result));
         }
 
+        public override FileSystemResult DehydrateFolder(string relativePath)
+        {
+            // Don't need to do anything here because the parent will reproject the folder.

```

Comment:
Do we need to ensure the parent folder is partial?  I am thinking about a scenario like this:

Repo:
```
src
  |
  -> Foo
    |
    -> bar
      |
      -> baz.txt
```

1. rmdir /s/q Foo
2. git checkout -- Foo
3. gvfs dehydrate --folders Foo\bar

When dehydrate is run "bar" will not be reprojected because Foo is a full folder.

Reply:
So this was a little messy because when you delete `Foo` that adds an entry to the modified paths and if you dehydrate any children that causes issues because the skip worktree bit will not be set for any children so nothing will be part of the projection even if we mark the parent folder as partial.  So what I ended up doing was preventing dehydration of folders that have a parent folder entry in the modified paths.

We could do more work here to maybe get this to work like add all the children that aren't being dehydrated to the the modified paths so we can remove the child specified for dehydration but to me that seems overkill and could introduce more risk and bugs.

Let me know what you think.

Reply:
> So what I ended up doing was preventing dehydration of folders that have a parent folder entry in the modified paths.

- Is this only a problem if the parent is in ModifiedPaths.dat?  Or is there a similar issue dehydrating folders that have children that are full folders?

- If a user *really* wanted Foo\bar dehydrated (in the above example) could they do so by dehydrating Foo?  

I think it makes sense to start with the restriction you mentioned and see what feedback we get from users. Ideally we'd have some manual work-around we could provide if someone manages to get stuck by this (even if it's as drastic as dehydrating the while repo) and so I'm interested in hearing your thoughts there.

Reply:
>Is this only a problem if the parent is in ModifiedPaths.dat? Or is there a similar issue dehydrating folders that have children that are full folders?

This is only a problem when there is a parent in the ModifiedPaths.dat.  If there are children that are full folders all the modified paths and placeholders for them are removed because it uses a starts with when removing entries.

>If a user really wanted Foo\bar dehydrated (in the above example) could they do so by dehydrating Foo?

Yes and I have a message that lets the user know that is what they would need to do to dehydrate.